test/controllers/ArrivalsControllerSpec.scala

```diff
@@ -16,27 +16,52 @@
 
 package controllers
 
+import controllers.actions.{AuthAction, FakeAuthAction}
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
 import akka.util.ByteString
+import config.AppConfig
 import org.scalatest.concurrent.ScalaFutures
+import org.scalatestplus.mockito.MockitoSugar
 import org.scalatest.{FreeSpec, MustMatchers, OptionValues}
 import org.scalatestplus.play.guice.GuiceOneAppPerSuite
 import play.api.http.HeaderNames
+import play.api.mvc.{AnyContentAsEmpty, BodyParsers}
 import play.api.test.{FakeHeaders, FakeRequest}
 import play.api.test.Helpers._
 
 import scala.xml.NodeSeq
+import services.XmlValidationService
+import uk.gov.hmrc.auth.core.AuthConnector
 
-class ArrivalsControllerSpec extends FreeSpec with MustMatchers with GuiceOneAppPerSuite with OptionValues with ScalaFutures {
-  def ctcFakeRequest() = FakeRequest(POST, routes.ArrivalsController.createArrivalNotification().url)
+import scala.concurrent.ExecutionContext.Implicits.global
 
-  def ctcFakeRequestXML() = FakeRequest(POST,
-    routes.ArrivalsController.createArrivalNotification().url)
-    .withHeaders(FakeHeaders(Seq(HeaderNames.CONTENT_TYPE -> "application/xml")))
+class ArrivalsControllerSpec extends FreeSpec with MustMatchers with GuiceOneAppPerSuite with OptionValues with ScalaFutures with MockitoSugar {
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
 
-  def ctcFakeRequestXML(body: NodeSeq) = FakeRequest(POST,
-    routes.ArrivalsController.createArrivalNotification().url)
-    .withHeaders(FakeHeaders(Seq(HeaderNames.CONTENT_TYPE -> "application/xml")))
-    .withXmlBody(body)
+  implicit def appConfig: AppConfig = app.injector.instanceOf[AppConfig]
+
+  val xmlValidation = new XmlValidationService
+
+  val cc = stubControllerComponents()
+
+  val config: AppConfig = app.injector.instanceOf[AppConfig]
+
+  val parser = app.injector.instanceOf[BodyParsers.Default]
+
+  val authConnector = mock[AuthConnector]
+  val fakeAuthAction = FakeAuthAction(authConnector, config, parser)
+
+  object TestArrivalsController extends ArrivalsController(cc, fakeAuthAction, xmlValidation)
```

Comment:
Rather than creating a test object that extends the controller we want to test, could we test the controller itself?

We should be able to remove lines 40 to 56, and instead have something that overrides `AuthAction` with our `FakeAuthAction` in the app.  This will be something like:

```
  override lazy val app = GuiceApplicationBuilder()
    .overrides(bind[AuthAction].to[FakeAuthAction])
    .build()
```

This will mean that whenever our tests run, they're using the `FakeAuthAction` instead of the real one, and we avoid calling off to auth etc.  There'll be a few more changes to the spec as well which are commented below